### DISCUSSION TITLE
How to display BottomSheet programmatically?

### DISCUSSION THREAD

**Discussion by tciuro** [LucasMucGH/BottomSheet#70] [Q&A] [answered]
I am showing a BottomSheet like so:

```
.bottomSheet(bottomSheetPosition: self.$bottomSheetPosition, options: [.tapToDismiss], content: {
...
})
```
After it has been dismissed, is there a way to display it again? In other words, is there a way to display it when say, a button has been tapped?

Thanks!

**Comment by pavelraspaev** [comment, marked as answer]
Hi!

You just need to set `var bottomSheetPosition` to `.middle` for open state and to `.hidden` for close state

Example:
```swift
Button {
    withAnimation {
        bottomSheetPosition = .middle
    }
} label: {
    Image("add")
        .resizable()
        .scaledToFit()
        .frame(width: 24, height: 24)
}
```

Hope it helps

**Reply by tciuro** [reply]
Awesome. Thanks so much!
